api/src/utils/sanitize-query.ts

```diff
@@ -1,18 +1,29 @@
 import type { Accountability, Aggregate, Filter, Query } from '@directus/types';
 import { parseFilter, parseJSON } from '@directus/utils';
 import { flatten, get, isPlainObject, merge, set } from 'lodash-es';
+import { getEnv } from '../env.js';
 import logger from '../logger.js';
 import { Meta } from '../types/index.js';
 
 export function sanitizeQuery(rawQuery: Record<string, any>, accountability?: Accountability | null): Query {
 	const query: Query = {};
 
+	const env = getEnv();
+
+	const hasMaxLimit =
+		'QUERY_LIMIT_MAX' in env &&
+		Number(env['QUERY_LIMIT_MAX']) >= 0 &&
+		!Number.isNaN(Number(env['QUERY_LIMIT_MAX'])) &&
+		Number.isFinite(Number(env['QUERY_LIMIT_MAX']));
+
 	if (rawQuery['limit'] !== undefined) {
 		const limit = sanitizeLimit(rawQuery['limit']);
 
 		if (typeof limit === 'number') {
-			query.limit = limit;
+			query.limit = limit === -1 && hasMaxLimit ? Number(env['QUERY_LIMIT_MAX']) : limit;
 		}
+	} else if (hasMaxLimit) {
+		query.limit = Math.min(Number(env['QUERY_LIMIT_DEFAULT']), Number(env['QUERY_LIMIT_MAX']));
```

Comment:
Is it intended that the `QUERY_LIMIT_DEFAULT` only applies if `QUERY_LIMIT_MAX` is defined?

Reply:
`QUERY_LIMIT_DEFAULT` is a fallback and gets applied here https://github.com/directus/directus/blob/e3d355f35327b62cb4bfbebf05962fde01465f47/api/src/database/run-ast.ts#L475-L490

If `limit` is not defined on the query then `sanitizeQuery` should not add it.

Reply:
Thank you! That makes sense!
Could we maybe make the doc a little clearer on this, for example like this:
`QUERY_LIMIT_DEFAULT` - The default query limit used **when `QUERY_LIMIT_MAX` is set and no limit is** defined in the API request.

Reply:
It does get applied in the quoted code, independent of `QUERY_LIMIT_MAX`

Reply:
Isn't that just for `o2m`?

Reply:
Hmmm looks like you're right, that seems to have been an oversight on my part 🤔 more hardcoded `100` limits should be using this environment var like https://github.com/directus/directus/blob/e3d355f35327b62cb4bfbebf05962fde01465f47/api/src/database/run-ast.ts#L253

Reply:
That latest change should properly set the `QUERY_LIMIT_DEFAULT` value if no limit is supplied